.circleci/config.yml

```diff
@@ -382,76 +370,98 @@ jobs:
 
   working_directory: ~/repo
 
-  build:
+  install:
     executor: node
     environment:
       # required to prevent ENOMEM errors
       LERNA_ARGS: --concurrency 1
     steps:
       - checkout
-      - detect_root_change
-      - cache_checksum
       - restore_cache:
-          name: Restore Yarn Package Cache
-          keys:
-            - yarn-packages-{{ checksum "yarn.lock" }}
-      - restore_node_modules_cache
+            name: Restore Yarn Package Cache
+            keys:
+              - yarn-packages-{{ checksum "yarn.lock" }}
       - install_node_dependencies
-      - save_cache:
-          name: Save Yarn Package Cache
-          key: yarn-packages-{{ checksum "yarn.lock" }}
-          paths:
-            - ~/.cache/yarn
-      - save_cache:
-          key: yarn-dependencies-{{ .Branch }}-{{ checksum "~/.tmp/checksumfiles/yarn.lock" }}
-          paths:
-            - node_modules
       - run: # Run PR Checks
-          name: Run PR Checks
-          command: yarn danger ci
+         name: Run PR Checks
+         command: yarn danger ci
+      - persist_to_workspace:
+          root: .
+          paths:
+            - '*'
+      - slack_notify_fail
+      - slack_notify_success
+
+  build:
+    executor: node
+    environment:
+      # required to prevent ENOMEM errors
+      LERNA_ARGS: --concurrency 2
+    steps:
+      - attach_workspace:
+          at: .
+      - detect_root_change
+      - restore_cache_dist_directories
       - build_packages
-      - run:
-          name: Run Bundlewatch checks
-          command: yarn bundlewatch
+      - save_cache_dist_directories
+      - build_storybook
+      - persist_to_workspace:
+          root: .
+          paths:
+            - '*'
       - lint_packages
       - slack_notify_fail
       - slack_notify_success
 
-  unit_tests:
+  bundlewatch:
     executor: node
     environment:
+      # required to prevent ENOMEM errors
       LERNA_ARGS: --concurrency 1
     steps:
-      - checkout
-      - detect_root_change
-      - cache_checksum
-      - restore_node_modules_cache
-      - install_node_dependencies
-      - restore_cache_dist_directories
-      - unit_integration_tests
+      - attach_workspace:
```

Comment:
By attaching the workspace that we persisted in the `build` job, Bundlewatch will have access to all the `dist` folders.